core/lib/node/src/task/mod.rs

```diff
@@ -0,0 +1,73 @@
+//! Tasks define the "runnable" concept of the node, e.g. something that can be launched and runs until the node
+//! is stopped.
+//!
+//! Task is normally defined by implementing two traits [`IntoZkSyncTask`], which acts like a constructor, and
+//! [`ZkSyncTask`], which provides an interface for `ZkSyncNode` to implement the task lifecycle.
+
+use futures::future::BoxFuture;
+
+use crate::node::{NodeContext, StopReceiver};
+
+/// Factory that can create a task.
+// Note: This have to be a separate trait, since `ZkSyncTask` has to be object-safe.
+pub trait IntoZkSyncTask: 'static + Send + Sync {
+    /// Unique name of the task.
+    const NAME: &'static str;
+
+    /// Config type for the task.
```

Comment:
that would be trivial to enforce if instead of Config and IntoZkSyncTask, you would have ConfigBuilder trait with a method `fn create(self, node NodeContext) -> Result<Box<ZksyncTask> TaskInitError>`.

Reply:
So, I've been thinking about it since yesterday, and am still torn. While probably it'll indeed look somewhat nicer (and also this trait would become object-safe, which would make it easier to make it async later), this would assume that each task _requires_ a config, but we might have tasks where `Config = ()` (e.g. if it some kind of an updater or logger or something that just interacts with resources provided by other tasks), and here it would result in unnecessary types.

In your opinion, is it a significant drawback?

Reply:
Also, if we'll go this route, we don't even have to mention configs anywhere, this may just be `TaskBuilder` trait -- so it's one less framework concept to think about.

Reply:
Although probably configless tasks wouldn't be a majority, so I guess it's OK. I'll implement it and will also check whether it will unblock async constructors.

Reply:
Types are cheap, and it is ok to have many types. From my experience, minimize-number-of-types mentality is what prevents people from writing type-safe code. Hence, I don't consider it a drawback really.

Reply:
Done, plus made everything async.